### PR TITLE
Updated to work on Mac.

### PR DESCRIPTION
**Pull request by emeryberger**

### CODE REVIEW

File: rust-support/chatdbg/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "chatdbg"
-version = "##VERSION##"                                 # Sync with pyproject.toml on release.
+version = "0.6.5"                                 # Sync with pyproject.toml on release.
```
Comment (nicovank) [inline rust-support/chatdbg/Cargo.toml:3]: These `##VERSION##` should stay as-is, they'll get auto-replaced 👍  The only one that needs changing is the main `pyproject.toml` one.

File: src/chatdbg/chatdbg_pdb.py [hunk 2]
```diff
@@ -14,6 +13,8 @@
 
 import IPython
 
+import chatdbg.custom_pdb as pdb
```
Comment (nicovank) [inline src/chatdbg/chatdbg_pdb.py:16]: Might be missing the `custom_pdb`.

**Review by nicovank — COMMENTED** [review]
From the error message it seems you are using 3.9 -- this is around where there were some semi-breaking changes to `pdb`. I think at some point we set the min supported Python version back to 3.10 since other packages we depend on need it (`IPython`, `ipdb`, possibly others).

Can you just upgrade to 3.10+ or is that not an option?

File: rust-support/chatdbg/Cargo.toml (hunk 1 above)
Comment (nicovank) [reply rust-support/chatdbg/Cargo.toml:3]: I realized the comment is now misleading -- will fix on master.

File: src/chatdbg/chatdbg_pdb.py (hunk 2 above)
Comment (emeryberger) [reply src/chatdbg/chatdbg_pdb.py:16]: Should be there?

**Comment by emeryberger** [comment]
The version of Python is tied to XCode, which for whatever reason is still at 3.9.

**Review by nicovank — COMMENTED** [review]
Some coverup test fail, just delete if tricky to fix.
```
rm test/test_coverup_19.py
rm test/test_coverup_20.py
rm test/test_coverup_32.py
rm test/test_coverup_46.py
```

EDIT: Nevermind, seems like you fixed it.

**Review by nicovank — APPROVED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -23,6 +23,8 @@ dependencies = [
   "PyYAML>=6.0.1", 
   "ipyflow>=0.0.130",
   "numpy>=1.26.3",
+  "httpcore>=1.0.5",
+  "httpx>=0.27.0",
```
[the remark this answers is not in the document]
Comment (emeryberger) [reply pyproject.toml:27]: In theory I agree, but we have no way of controlling those packages and they are not forcing an upgrade on their own. It leads to a pretty confusing error...

[the remark this answers is not in the document]
Comment (nicovank) [reply pyproject.toml:27]: Sure if it improves our error message 👍

**Pull request merged**